excavator/utils.py

```diff
@@ -80,7 +81,49 @@ def env_int(name, required=False, default=empty):
     value = get_env_value(name, required=required, default=default)
     if value is empty:
         raise ValueError(
-            "`env_int` requires either a default value to be specified, or, for "
+            "`env_int` requires either a default value to be specified, or for "
             "the variable to be present in the environment"
         )
     return int(value)
+
+
+def _parse_timestamp(timestamp_str):
+    timestamp = float(timestamp_str)
+    return datetime.datetime.fromtimestamp(timestamp)
+
+
+def _parse_iso8601(iso8601_str):
+    try:
+        import iso8601
+    except ImportError:
+        raise ImportError(
+            'Parsing iso8601 datetime strings requires the iso8601 library'
+        )
+    return iso8601.parse_date(iso8601_str)
+
+
+_formats = {
+    'timestamp': _parse_timestamp,
+    'iso8601': _parse_iso8601,
+}
+
+
+def env_datetime(name, default=empty, required=False, fmt='timestamp'):
```

Comment:
> Why not just have separated definitions for `env_timestamp` and `env_iso8601`

Interesting.

**things I like about it**
- it's explicit about what it does

**things I don't like**
- `env_timestamp` does not return a timestamp but a `datetime`.  Not entirely intuitive.
- `end_iso8601` implies some knowledge of what `iso8601` actually is.
- it's a lot easier to remember `env_datetime`

**things I don't like about the `fmt` kwarg**
- You have to know about the api to use it (as in you have to know about what formats are supported)

What if I made it required to explicitly state the `fmt`.

```python
>>> excavator.env_datetime('SOME_VARIABLE')
ValueError("Must specify a format with the `fmt` keyword.  Supported formats are 'timestamp' and 'iso_8601'")
```